src/formula.py

```diff
@@ -168,6 +160,42 @@ def install(self) -> None:
                     shutil.copyfileobj(src, dst)
                 output_path.chmod(info.mode)
 
+# something split into e.g. bin/, lib/, include/
+class UnixPackageFormula(Formula):
+    archive_url: str
+
+    @contextlib.contextmanager
+    def _read_archive(self, filename: str, fp: BinaryIO) -> Iterator[Archive]:
+        archive_type = file_archive_type(filename)
+        if archive_type == "zip":
+            yield ZipArchive(io.BytesIO(fp.read()))
+        elif archive_type == "tar":
+            yield TarArchive(io.BytesIO(fp.read()))
+        else:
+            raise ValueError(f"invalid archive type: {archive_type!r}")
+
+    def install(self) -> None:
+        download_url = self._eval_member("archive_url")
+        self.log('fetching file at "%s"', download_url)
+        with urllib.request.urlopen(download_url) as response, self._read_archive(download_url, response) as archive:
+            for item in archive.members():
+                output_path = Path('/usr/local') / item
```

Comment:
Would be good to respect `install_to`; makes it easy to test the formula locally.

Reply:
oh, that makes sense